src/packages/core/src/server.ts

```diff
@@ -149,25 +182,36 @@ export default class Server {
     if (this.#status === Status.opening) {
       // if opening
       throw new Error(`Cannot close server while it is opening.`);
-    } else if (this.#status & Status.closed) {
-      // if closed or closing
-      throw new Error(`Server is already closed or closing.`);
+    } else if ((this.#status & Status.closingOrClosed) !== 0) {
+      // if closing or closed
+      throw new Error(`Server is already closing or closed.`);
     }
 
-    const _listenSocket = this.#listenSocket;
     this.#status = Status.closing;
+
+    // clean up the websocket objects
+    const _listenSocket = this.#listenSocket;
     this.#listenSocket = void 0;
```

Comment:
I think this should be set to `null` here, to reflect your type change.

```suggestion
    this.#listenSocket = null;
```